spikeinterface/qualitymetrics/misc_metrics.py

```diff
@@ -488,6 +485,207 @@ def compute_amplitudes_cutoff(waveform_extractor, peak_sign='neg',
+        The direction along which drift metrics are estimated, by default 'y'
+    min_fraction_valid_intervals : float, optional
+        The fraction of valid (not nan) position estimates to estimate drifts.
+        E.g., if 0.5 at least 50% of estimated positions in the intervals need to be valid,
+        otherwise drift metrics are set to None, by default 0.5
+    return_positions : bool, optional
+        If True, median positions are returned (for debugging), by default False
+
+    Returns
+    -------
+    maximum_drift : dict
+        The maximum drift in um
+    cumulative_drift : dict
+        The cumulative drift in um
+    """
+    res = namedtuple("drift_metrics", ['maximum_drift', 'cumulative_drift'])
+
+    if waveform_extractor.is_extension("spike_locations"):
+        locs_calculator = waveform_extractor.load_extension("spike_locations")
+        spike_locations = locs_calculator.get_data(outputs="concatenated")
+    else:
+        warnings.warn("The drift metrics require the `spike_locations` waveform extension. "
+                      "Use the `postprocessing.compute_spike_locations()` function. "
+                      "Drift metrics will be set to NaN")
+        empty_dict = {unit_id: np.nan for unit_id in waveform_extractor.unit_ids}
+        if return_positions:
+            return res(empty_dict, empty_dict), np.nan
+        else:
+            return res(empty_dict, empty_dict)
+
+    recording = waveform_extractor.recording
+    sorting = waveform_extractor.sorting
+    unit_ids = waveform_extractor.unit_ids
+    maximum_drift = {}
+    cumulative_drift = {}
+    maximum_drift_segs = {unit_id: [] for unit_id in unit_ids}
+    cumulative_drift_segs = {unit_id: [] for unit_id in unit_ids}
+    interval_samples = int(interval_s * waveform_extractor.sampling_frequency)
+    assert direction in spike_locations.dtype.names, (
+        f"Direction {direction} is invalid. Available directions: "
+        f"{spike_locations.dtype.names}"
+    )
+
+    # if a segment is shorter than 2 * interval_s, we set the drift metrics to -1
+    for segment_index in range(recording.get_num_segments()):
+        seg_length = recording.get_num_samples(segment_index)
+        if  seg_length <= 2 * interval_samples:
+            warnings.warn(f"Segment {segment_index} is too short to compute drift metrics "
+                          f"with interval_s of {interval_s}. It needs to be at least {2*interval_s}s long")
+            for unit_id in unit_ids:
+                maximum_drift_segs[unit_id].append(-1)
+                cumulative_drift_segs[unit_id].append(-1)
+        else:
+            num_bin_edges = seg_length // interval_samples + 1
+            bins = np.arange(num_bin_edges) * interval_samples
+            spike_vector = sorting.to_spike_vector()
+
+            # compute median positions (if less than min_spikes_per_interval, median position is 0)
+            median_positions = np.nan * np.zeros((len(unit_ids), num_bin_edges - 1))
```

Comment:
Here the median positions are computed at every segment.
This is an important choice!!!
If segment are consecutive (because a short pause) in thet case the median positions should global no ?

Reply:
We don't have that information. I think it's better to treat each segment as separate